### PR TITLE
KYLIN-3588, fix potentially duplicate putting in RemoveBlackoutRealiz…

### PR DESCRIPTION
**Pull request by tttMelody**
…ationsRule#getFilterImpl

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: query/src/main/java/org/apache/kylin/query/routing/rules/RemoveBlackoutRealizationsRule.java [hunk 1]
```diff
@@ -58,14 +58,16 @@ public static boolean accept(IRealization real) {
     
     private static IRealizationFilter getFilterImpl(KylinConfig conf) {
         IRealizationFilter filter = filters.get(conf);
-        if (filter == null) {
-            try {
-                Class<? extends IRealizationFilter> clz = ClassUtil.forName(conf.getQueryRealizationFilter(), IRealizationFilter.class);
-                filter = clz.getConstructor(KylinConfig.class).newInstance(conf);
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+        synchronized (RemoveBlackoutRealizationsRule.class) {
```
[the remark this answers is not in the document]
Comment (tttMelody) [reply query/src/main/java/org/apache/kylin/query/routing/rules/RemoveBlackoutRealizationsRule.java]: Thanks shaofeng! I'll refine this.

**Comment by tttMelody** [comment]
Thanks @shaofengshi ! The opt is done.

**Pull request merged**